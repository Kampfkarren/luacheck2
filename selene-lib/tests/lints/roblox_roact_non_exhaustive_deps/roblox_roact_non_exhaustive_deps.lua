local React

local notreactive1 = {}

local function Component1()
    local reactive1 = {}
    local reactive2 = {}

    React.useEffect(function()
        local allowed = notreactive1
        local notallowed = reactive1
        print(reactive2)
    end, {})
end

local function Component2()
    local reactive1 = {}

    React.useEffect(function()
        allowed(reactive1, `{notreactive1}`)
    end, { reactive1 })
end

local function Component3()
    local reactive1 = {}
    local reactive2 = {}

    React.useEffect(function()
        notreactive1(reactive1(reactive2()))
    end, depArray(reactive1))
end

local function Component5()
    local reactive1 = {}

    React.useEffect(function()
        local notreactive = function()
            if true then
                local notallowed = reactive1
            end
        end
        notreactive()
    end, {})
end

local function Component6()
    React.useEffect(function() end, { error, notreactive1 })
end

local function Component7()
    local reactive1 = {}

    React.useEffect(function()
        print(reactive1)
    end, { notreactive1 })
end

local function MakeComponent()
    local notreactive = {}

    local function Component()
        local reactive1 = {}

        React.useEffect(function()
            local allowed = notreactive
            local allowed = reactive1
        end, { reactive1 })
    end
end
