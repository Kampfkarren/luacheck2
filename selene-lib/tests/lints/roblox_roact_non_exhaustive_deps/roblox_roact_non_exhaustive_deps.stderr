error[roblox_roact_non_exhaustive_deps]: react hook useEffect has missing dependencies: 'reactive1' and 'reactive2'
   ┌─ roblox_roact_non_exhaustive_deps.lua:13:10
   │
13 │     end, {})
   │          ^^
   │
   = help: either include them or remove the dependency array

error[roblox_roact_non_exhaustive_deps]: react hook useEffect has missing dependency: 'reactive2'
   ┌─ roblox_roact_non_exhaustive_deps.lua:30:10
   │
30 │     end, depArray(reactive1))
   │          ^^^^^^^^^^^^^^^^^^^
   │
   = help: either include it or remove the dependency array

error[roblox_roact_non_exhaustive_deps]: react hook useEffect has missing dependency: 'reactive1'
   ┌─ roblox_roact_non_exhaustive_deps.lua:43:10
   │
43 │     end, {})
   │          ^^
   │
   = help: either include it or remove the dependency array

error[roblox_roact_non_exhaustive_deps]: react hook useEffect has unnecessary dependencies: 'error' and 'notreactive1'
   ┌─ roblox_roact_non_exhaustive_deps.lua:47:37
   │
47 │     React.useEffect(function() end, { error, notreactive1 })
   │                                     ^^^^^^^^^^^^^^^^^^^^^^^
   │
   = help: either exclude them or remove the dependency array
   = outer scope variables like 'error' aren't valid dependencies because mutating them doesn't re-render the component

error[roblox_roact_non_exhaustive_deps]: react hook useEffect has missing dependency: 'reactive1'
   ┌─ roblox_roact_non_exhaustive_deps.lua:55:10
   │
55 │     end, { notreactive1 })
   │          ^^^^^^^^^^^^^^^^
   │
   = help: either include it or remove the dependency array

error[roblox_roact_non_exhaustive_deps]: react hook useEffect has unnecessary dependency: 'notreactive1'
   ┌─ roblox_roact_non_exhaustive_deps.lua:55:10
   │
55 │     end, { notreactive1 })
   │          ^^^^^^^^^^^^^^^^
   │
   = help: either exclude it or remove the dependency array
   = outer scope variables like 'notreactive1' aren't valid dependencies because mutating them doesn't re-render the component

